File: week_1/q1.m
%3PO: Tutorial 1, Question 1
%Genevieve Clifford (1779290), University of Birmingham
%sxc1065@student.bham.ac.uk / genevieve@becquerel.me

clear all; close all; clc;

P=[20;30];
A=[1 2;1 1];
b=[120;90];
lb=[0;0];
ub=[70;50];

[x,fval]=linprog(-P,A,b,[],[],lb,ub);

fval=-fval;

formatSpec='A profit of £%4.2f is to be made by producing %2.0f low-cost and %2.0f high-cost units respectively';
fprintf(formatSpec,fval,x(1),x(2))

File: week_1/q1.py
"""
3PO: Tutorial 1, Question 1
Genevieve Clifford (1779290), University of Birmingham
sxc1065@student.bham.ac.uk / genevieve@becquerel.me
"""

import numpy as np
from scipy.optimize import linprog

A_ub=np.array([
    [1,0],
    [0,1],
    [1,2],
    [1,1]
])

b_ub=np.array([70,50,120,90])

c=np.array([-20,-30])

res=linprog(c,A_ub=A_ub,b_ub=b_ub,bounds=(0,None))

print('A profit of £',-round(res.fun,0),' is to be made by producing ',round(res.x[0],0),
' low-cost and ',round(res.x[1],0),' high-cost units respectively')